calblueprint/wecaresolar
# "Add to Home Screen" Functionality

## 🖋 PR Description

Allows the app to be installed on mobile devices through an automatic "Add to Home Screen" prompt when the user first visits the website. A shortcut to the home page will show up as an icon on the user's home screen, and the app will run in fullscreen without showing the default browser UI.

### 🤳Screenshots (if applicable)
* *On computer*: https://www.loom.com/share/2e09fda5c28d470cab3d918479fe0eff
* *On Android*: https://www.loom.com/share/36165f0fdae54ec1b8ecddeaff32280a

### 🐞 Bug Report
* There are very specific requirements for the "Add to Home Screen" prompt to show, and if they aren't met it will just silently fail without error messages. Specifically, we need to have icons of size 192x192 and 512x512 (@dfangshuo maybe we should make some nice looking icons of this size? right now i'm just using a scaled-up version of the WCS icon) and a valid service worker. See Notes section for details

### Notes
* In the future, we should look into displaying installation prompts and tracking whether the user has installed the app (for analytics and styling): https://web.dev/customize-install/
* Requirements for showing prompt: https://stackoverflow.com/questions/51586398/pwa-add-to-home-screen-not-showing-popup

### 🧪 Testing
**Testing on a mobile device**
* Go to [this link](https://we-care-solar-c3e5f--pr20-kevin-add-to-home-sc-j8ap8csh.web.app/) on an Android simulator
* You should receive a prompt to add the app to your home screen. Click "Add"
* Exit Chrome and look for a new app named "We Care Solar" on your home screen
* Open it and you should be able to see our website in fullscreen mode!

**Testing on Chrome on a laptop/computer**
* Go to [this link](https://we-care-solar-c3e5f--pr20-kevin-add-to-home-sc-j8ap8csh.web.app/) 
* Click on the small "+" icon on the very right side of your address bar, then click "Install"
* The app will be saved in a "Chrome Apps" directory on your computer and should automatically open up

cc: @joelenelatief
